src/MessageHandler.ts

```diff
@@ -1,90 +1,81 @@
 import Discord from 'discord.js';
 import './discord/Message';
 
-import QueueItem from './queue/QueueItem';
+import * as Commands from './commands/Commands';
+
+import DatabaseAdapter from './db/DatabaseAdapter';
 import SoundQueue from './queue/SoundQueue';
-import Util from './Util';
 
 export default class MessageHandler {
-  private readonly prefix: string;
   private readonly queue: SoundQueue;
+  private readonly db: DatabaseAdapter;
+  private readonly prefix: string;
 
-  constructor(queue: SoundQueue, prefix: string) {
+  constructor(queue: SoundQueue, db: DatabaseAdapter, prefix: string) {
     this.queue = queue;
+    this.db = db;
     this.prefix = prefix;
   }
 
   public handle(message: Discord.Message) {
     if (message.isDirectMessage()) return;
     if (!message.hasPrefix(this.prefix)) return;
-    if (Util.isIgnoredUser(message.author)) return;
+
+    // @REVIEW Move this check to User.isIgnored?
```

Comment:
The way you do it here is correct IMHO. Maybe DatabaseAdapter is not the best name but I fail to come up with a better one right now. You could have more specific interfaces that DatabaseAdapter implements and then each requiring class only requires the interface it needs. But right now that is unnecessary imho and I would leave it as is.